### PR TITLE
Fix detection of nested types

### PR DESCRIPTION
**Pull request by carlossanlop**
After fixing the MSBuild problem, the unit tests uncovered a problem when attempting to detect nested types, specifically, delegates. Turns out the way symbols were being visited before was not a proper way of ensuring nested types would also be visited. Instead, I added a CSharpVisitor instance and now all the symbols get pre-collected, and then paired up with the docs types afterwards.

The tests now pass.

### CODE REVIEW

File: src/PortToTripleSlash/src/libraries/ToTripleSlashPorter.cs [hunk 1]
```diff
@@ -42,286 +40,206 @@ public ToTripleSlashPorter(Configuration config)
...
             {
-                throw new Exception($"No symbols found for '{docsType.TypeName}'");
+                string ns = Path.GetFileNameWithoutExtension(projectPath).ToUpperInvariant();
+                projectNamespace = ns;
+
+                // Skip looking in projects whose namespace that were explicitly excluded or not explicitly included
+                // The only exception is System.Private.CoreLib, which we should always explore
+                if (projectNamespace != _systemPrivateCoreLib)
+                {
+                    if (!_config.IncludedNamespaces.Any(x => x.ToUpperInvariant().StartsWith(ns)) ||
```
Comment (smasher164) [inline src/PortToTripleSlash/src/libraries/ToTripleSlashPorter.cs:158]: if either of these is true or both of these is true?

**Review by smasher164 — APPROVED** [review]
Looks good to me!

**Comment by carlossanlop** [comment]
Need to fix this in the yml. I'll send it in a separate PR:
```
Error: /usr/share/dotnet/sdk/6.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(144,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 7.0.  Either target .NET 6.0 or lower, or use a version of the .NET SDK that supports .NET 7.0. [/home/runner/work/api-docs-sync/api-docs-sync/src/PortToTripleSlash/src/app/PortToTripleSlash.csproj]
```

**Comment by carlossanlop** [comment]
Closing/reopening to auto rebase.

**Pull request merged**